### PR TITLE
feat(cmd/gnoland): cli implement --home

### PR DESCRIPTION
**Pull request by albttx**
This implement the point :two: and :four: in my wishlist (#2102)

This PR implement a new flag `--home` and remove multiple flags like: `--output-path`  or `--data-dir`, `--config-path` which could be confusing while using `gnoland` binary.

All config is under the directory specified by `--home`

⚠️  genesis is by default under `{{ --home }}/genesis.json` 🎉 

### Examples: 

- `gnoland genesis generate --output-path $GENESIS_FILE`

```bash
$ gnoland genesis generate
Genesis successfully generated at ./gnoland-data/genesis.json

$ gnoland genesis generate --genesis /tmp/genesis.json                                                                                               ✔
Genesis successfully generated at /tmp/genesis.json
```

- `gnoland config set --config-path ./gnoland-data/config/config.toml rpc.laddr "tcp://0.0.0.0:26657"`

```bash
$ gnoland config set rpc.laddr "tcp://0.0.0.0:26657"

$ gnoland config set  --home ~/gnoland-data rpc.laddr "tcp://0.0.0.0:26657"
```

### CODE REVIEW

File: gno.land/cmd/gnoland/root.go [hunk 1]
```diff
@@ -2,32 +2,49 @@ package main
...
-
 func main() {
 	cmd := newRootCmd(commands.NewDefaultIO())
 
 	cmd.Execute(context.Background(), os.Args[1:])
 }
 
+type rootCfg struct {
+	homeDir homeDirectory
+}
```
Comment (thehowl) [inline gno.land/cmd/gnoland/root.go:21]: please, for flag structs, don't nest them. They can have a helper type like the one you created, but there shouldn't be 3 levels of nesting

File: gno.land/cmd/gnoland/home.go
Comment (thehowl) [inline gno.land/cmd/gnoland/home.go:1]: Enough, I don't want any more files in this damn directory

Put this in `root.go`

**Review by thehowl — COMMENTED** [review]
After doing a node deployment myself, I agree this is a good way to go, as for using genesis.json I've had to do quite a bit of a sing and dance to get it working.

cc @sw360cab for his first review :)

File: gno.land/cmd/gnoland/root.go (hunk 1 above)
Comment (albttx) [reply gno.land/cmd/gnoland/root.go:47]: I'm not sure to understand how you want it not nested ?

Comment (thehowl) [reply gno.land/cmd/gnoland/root.go:47]: Yeah, I don't know, I had a bit of a think and discussion with Milos and I'm not sure how this should work. cc @zivkovicmilos

I don't like `rootCfg` :shrug:

File: gno.land/cmd/gnoland/home_test.go
Comment (thehowl) [inline gno.land/cmd/gnoland/home_test.go:1]: please move this file as well

**Comment by albttx** [comment]
@thehowl @zivkovicmilos  it's rebased and i fixed your reviews

File: gno.land/cmd/gnoland/root.go [hunk 2]
```diff
@@ -40,3 +58,39 @@ func newRootCmd(io commands.IO) *commands.Command {
...
 	return cmd
 }
+
+type homeDirectory struct {
+	homeDir     string
+	genesisFile string
+}
+
+func (h homeDirectory) Path() string       { return h.homeDir }
+func (h homeDirectory) ConfigDir() string  { return h.Path() + "/config" }
```
Comment (moul) [inline gno.land/cmd/gnoland/root.go:68]: Use `filepath.Join`

File: gno.land/cmd/gnoland/secrets_verify_test.go [hunk 3]
```diff
@@ -184,24 +126,24 @@ func TestSecrets_Verify_All_Missing(t *testing.T) {
 			initArgs := []string{
 				"secrets",
 				"init",
-				"--data-dir",
-				tempDir,
+				"--home",
```
Comment (moul) [inline gno.land/cmd/gnoland/secrets_verify_test.go:129]: If we use a single flag, I think we should use `—data`

File: gno.land/pkg/keyscli/root.go [hunk 4]
```diff
@@ -19,7 +19,6 @@ func NewRootCmd(io commands.IO, base client.BaseOptions) *commands.Command {
 			ShortUsage: "<subcommand> [flags] [<arg>...]",
 			LongHelp:   "Manages private keys for the node",
 			Options: []ff.Option{
-				ff.WithConfigFileFlag("config"),
```
Comment (moul) [inline gno.land/pkg/keyscli/root.go:22]: Why?

File: tm2/pkg/crypto/keys/client/root.go [hunk 5]
```diff
@@ -28,7 +28,6 @@ func NewRootCmdWithBaseConfig(io commands.IO, base BaseOptions) *commands.Comman
 			ShortUsage: "<subcommand> [flags] [<arg>...]",
 			LongHelp:   "Manages private keys for the node",
 			Options: []ff.Option{
-				ff.WithConfigFileFlag("config"),
```
Comment (moul) [inline tm2/pkg/crypto/keys/client/root.go:31]: Why?

**Comment by moul** [comment]
Adding genesis, config, and secret commands to the gnoland command was not the best move. They belong in a separate tool like `contribs/gnolandtools`, with fewer files involved. Or maybe they should just not exist.

The gnoland binary is meant to be a stable, finished product, not an SDK for running multiple blockchains. We're not going to be tweaking genesis files daily. Gno.land will have its mainnet, testnets, and a local setup that's either lazy or uses the latest snapshot. Other projects might create their own lands, but they won't be using the gnoland binary for that. Most of the configuration will be managed by the chain too.

In other words, 99.9% of people will use lazy or download a ready-to-start state.

Here's what we really need:
- A solid `--lazy` option for easy setup.
- A snapshot feature for gnodev. (100x better than `gnoland genesis xxx` commands)
- A reliable editor to patch the genesis file.
- Clear documentation and basic tool for managing secret keys.
- Keeping gnoland light.

**Comment by albttx** [comment]
@moul I understand what you means, but i really think those commands are required from an ops perspective.

Creating a genesis file for a multi node cluster with `gnoland start --lazy` -> `ctrl+c` is just impossible.

You already told me this commands will come in a `gno-sdk` which i believe could be the right place, maybe for the moment let's keep it inside of the `gnoland` binary, but once ready,  we could move it into this `gno-sdk` ?

about the rename from `--home` into `--data`, i believe it could lead to confusions, because the folder contains datas, configurations, secrets...

File: tm2/pkg/crypto/keys/client/root.go (hunk 5 above)
Comment (albttx) [reply tm2/pkg/crypto/keys/client/root.go:31]: because `--config` is replaced with `--home`, which is not a file

File: gno.land/cmd/gnoland/secrets_verify_test.go (hunk 3 above)
Comment (albttx) [reply gno.land/cmd/gnoland/secrets_verify_test.go:129]: I believe it could lead to confusions, because the folder contains datas, configurations, secrets...

**Comment by thehowl** [comment]
It's definitely very stale.
